packages/sources/tradingeconomics-test/src/endpoint/price-router.ts

```diff
@@ -0,0 +1,98 @@
+import { RoutingTransport } from '@chainlink/external-adapter-framework/transports/meta'
+import { wsTransport } from './price-ws'
+import { customSettings } from '../config'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { httpTransport } from './price'
+import {
+  PriceEndpoint,
+  PriceEndpointInputParameters,
+  PriceEndpointParams,
+} from '@chainlink/external-adapter-framework/adapter'
+
+export const inputParameters: PriceEndpointInputParameters = {
+  base: {
+    aliases: ['from', 'coin', 'fsym'],
+    description: 'The symbol of symbols of the currency to query',
+    required: true,
+    type: 'string',
+  },
+  quote: {
+    aliases: ['to', 'market', 'tsym'],
+    description: 'The symbol of the currency to convert to',
+    required: true,
+    type: 'string',
+  },
+} as const
+
+export interface ProviderRequestBody {
+  events: string
+  symbols: string
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: PriceEndpointParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: ProviderRequestBody
+    ResponseBody: ProviderResponseBody[]
+  }
+}
+
+export interface ProviderResponseBody {
+  Symbol: string
+  Ticker: string
+  Name: string
+  Country: string
+  Date: string
+  Type: string
+  decimals: number
+  state: string
+  Last: number
+  Close: number
+  CloseDate: string
+  MarketCap: number | null
+  URL: string
+  Importance: number
+  DailyChange: number
+  DailyPercentualChange: number
+  WeeklyChange: number
+  WeeklyPercentualChange: number
+  MonthlyChange: number
+  MonthlyPercentualChange: number
+  YearlyChange: number
+  YearlyPercentualChange: number
+  YTDChange: number
+  YTDPercentualChange: number
+  day_high: number
+  day_low: number
+  yesterday: number
+  lastWeek: number
+  lastMonth: number
+  lastYear: number
+  startYear: number
+  ISIN: string | null
+  frequency: string
+  LastUpdate: string
+}
+
+export const routingTransport = new RoutingTransport<EndpointTypes>(
+  {
+    REST: httpTransport,
+    WS: wsTransport,
+  },
+  (_, adapterConfig) => {
+    if (adapterConfig.WS_ENABLED) {
+      return 'WS'
+    }
+    return 'REST'
+  },
+)
+
+export const endpoint = new PriceEndpoint<EndpointTypes>({
+  name: 'price',
+  transport: routingTransport,
```

Comment:
I think we may need to add a few aliases that ensure backwards compatibility here since we fail on unmatched endpoints instead of fall back to the default in v3. From what I'm gathering, I think we'll need these since we use the same endpoint for all data in TradingEconomics.
```suggestion
  aliases: ['stock', 'forex'],
  transport: routingTransport,
```

Reply:
Actually I think we may want to split `stock` off into it's own endpoint. It looks like it has a difference in input parameters than `crypto` and `forex` since it only uses `base`.